### PR TITLE
Minor updates to API docs in README to reflect current main

### PR DESCRIPTION
**Pull request by ellisonbg**
I have gone through and audited the API docs in the README and updated them to reflect the current state of the code/APIs in main. May have missed a few things, but this should be closer.

### CODE REVIEW

**Comment by ellisonbg** [comment]
Fixes #231

**Review by dmonad — COMMENTED** [review]
Hi @ellisonbg, thanks for working through the API! 

I added a few comments. I'm a bit hesitant to put new stuff into the API documentation because I need to maintain the described functions and keep them running. I also make sure to have at least a few tests that check each documented function. 

The *snapshot* feature is not yet fully supported. I don't feel confident that I will keep it as-is, so I don't want to document it. 

I'd like to merge the rest of the additions.

My general rule is that undocumented features shouldn't be used. It is probably a bit problematic that you can see the full API from the typescript declarations though. But I'm kinda happy that there are only so few undocumented features.

**Comment by ellisonbg** [comment]
Thanks Kevin, I will address your feedback at some point soon.

File: README.md [hunk 1]
```diff
@@ -387,16 +389,20 @@ YTextEvents compute changes as deltas.
   </p>
   <pre>const ytext = new Y.Text()</pre>
   <dl>
-    <b><code>insert(index:number, content:string, [formattingAttributes:Object&lt;string,string&gt;])</code></b>
```
[the remark this answers is not in the document]
Comment (aryzing) [reply README.md:390]: If `TextAttributes` can be defined, that would be great. If we're not ready to define it yet, possibly `Record<string, string>` would be a more semantic type.

**Comment by notgiorgi** [comment]
Hi @ellisonbg thanks for working on this.

You can add this one as well: https://github.com/yjs/yjs/commit/034463798df62fd2a1950179c19a6f06bb87caf6

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (dmonad) [reply README.md:390]: I kept using Object<string, string> because that is what I use all over the codebase.

**Comment by dmonad** [comment]
Thanks for this @ellisonbg & @notgiorgi I just implemented the suggestions in a different commit.

**Pull request closed without merging**